src/components/createActivityForm/CreateActivityForm.js

```diff
@@ -0,0 +1,158 @@
+import { nanoid } from 'nanoid'
+import styled from 'styled-components'
+
+export default function CreateActivityForm({ activities, setActivities }) {
+  function createNewActivity({
+    name,
+    description,
+    street,
+    zipCode,
+    city,
+    country,
+    openingHours,
+    website,
+    isFreeOfCharge,
+  }) {
+    const newActivity = [
+      ...activities,
+      {
+        id: nanoid(),
+        name: name,
+        description: description,
+        street: street,
+        city: city,
+        zipCode: zipCode,
+        country: country,
+        openingHours: openingHours,
+        website: website,
+        isFreeOfCharge: isFreeOfCharge,
+      },
+    ]
+    setActivities(newActivity)
+  }
```

Comment:
```suggestion
  function createNewActivity(newActivity) {
    const newActivities = [
      ...activities,
      newActivity,
    ]
    setActivities(newActivities)
  }
```

Short and sweet :) There is no need to destructure the new activity and directly recreate it. Also try to keep the naming consistent.

Reply:
I think this function can be pulled up into App.js, the form only has to deal with the new activity and not all the activities that exist

Reply:
This is so much better. Thanks a lot. Changed it.